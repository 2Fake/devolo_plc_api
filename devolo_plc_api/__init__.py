"""The devolo PLC API."""
from importlib.metadata import PackageNotFoundError, version

from .device import Device

try:
    __version__ = version("devolo_plc_api")
except PackageNotFoundError:
    # package is not installed - e.g. pulled and run locally
    __version__ = "0.0.0"

__all__ = ["Device", "__version__"]
