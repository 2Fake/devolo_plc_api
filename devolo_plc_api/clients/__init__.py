"""Clients used to communicate with devolo devices."""
from .protobuf import Protobuf

__all__ = ["Protobuf"]
