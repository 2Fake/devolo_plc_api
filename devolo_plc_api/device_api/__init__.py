"""The devolo device API."""
from .deviceapi import DeviceApi

SERVICE_TYPE = "_dvl-deviceapi._tcp.local."

__all__ = ["DeviceApi"]
