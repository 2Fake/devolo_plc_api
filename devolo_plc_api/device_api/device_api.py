from .wifi_network.wifi_network import WifiNetwork
from ..helper.zeroconf import get_token


class DeviceApi:
    def __init__(self, ip, session):
        self.token, self.features = get_token(ip, "_dvl-deviceapi._tcp.local.")
        feature_mapping = {"wifi1": WifiNetwork}

        for feature in self.features:
            try:
                setattr(self, feature, feature_mapping.get(feature)(ip, session, 14791, self.token))
            except TypeError:
                continue