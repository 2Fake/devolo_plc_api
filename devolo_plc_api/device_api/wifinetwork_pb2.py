# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: wifinetwork.proto
"""Generated protocol buffer code."""
from google.protobuf.internal import enum_type_wrapper
from google.protobuf import descriptor as _descriptor
from google.protobuf import descriptor_pool as _descriptor_pool
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()




DESCRIPTOR = _descriptor_pool.Default().AddSerializedFile(b'\n\x11wifinetwork.proto\x12\ndevice.api\"E\n\x19WifiDeviceModeGetResponse\x12(\n\x04mode\x18\x01 \x01(\x0e\x32\x1a.device.api.WifiDeviceMode\"\xa9\x01\n\x11WifiDeviceModeSet\x12(\n\x04mode\x18\x01 \x01(\x0e\x32\x1a.device.api.WifiDeviceMode\x12\x30\n\tfh_params\x18\x02 \x01(\x0b\x32\x1d.device.api.WifiParametersSet\x12\x38\n\tbh_params\x18\x03 \x01(\x0b\x32%.device.api.WifiRepeaterParametersSet\"C\n\x19WifiDeviceModeSetResponse\x12&\n\x06result\x18\x01 \x01(\x0e\x32\x16.device.api.WifiResult\".\n\x11WifiParametersSet\x12\x0c\n\x04ssid\x18\x01 \x01(\t\x12\x0b\n\x03key\x18\x02 \x01(\t\"C\n\x19WifiParametersSetResponse\x12&\n\x06result\x18\x01 \x01(\x0e\x32\x16.device.api.WifiResult\"w\n\x12WifiGuestAccessSet\x12\x0e\n\x06\x65nable\x18\x01 \x01(\x08\x12\x10\n\x08\x64uration\x18\x02 \x01(\r\x12\x0c\n\x04ssid\x18\x03 \x01(\t\x12\x0b\n\x03key\x18\x04 \x01(\t\x12$\n\x03wpa\x18\x05 \x01(\x0e\x32\x17.device.api.WifiWpaMode\"D\n\x1aWifiGuestAccessSetResponse\x12&\n\x06result\x18\x01 \x01(\x0e\x32\x16.device.api.WifiResult\"\x82\x01\n\x12WifiGuestAccessGet\x12\x0f\n\x07\x65nabled\x18\x01 \x01(\x08\x12\x1a\n\x12remaining_duration\x18\x02 \x01(\r\x12\x0c\n\x04ssid\x18\x03 \x01(\t\x12\x0b\n\x03key\x18\x04 \x01(\t\x12$\n\x03wpa\x18\x05 \x01(\x0e\x32\x17.device.api.WifiWpaMode\"\xe9\x01\n\x12WifiNeighborAPsGet\x12\x43\n\x0cneighbor_aps\x18\x01 \x03(\x0b\x32-.device.api.WifiNeighborAPsGet.NeighborAPInfo\x1a\x8d\x01\n\x0eNeighborAPInfo\x12\x13\n\x0bmac_address\x18\x01 \x01(\t\x12\x0c\n\x04ssid\x18\x02 \x01(\t\x12\"\n\x04\x62\x61nd\x18\x03 \x01(\x0e\x32\x14.device.api.WifiBand\x12\x0f\n\x07\x63hannel\x18\x04 \x01(\r\x12\x0e\n\x06signal\x18\x05 \x01(\x05\x12\x13\n\x0bsignal_bars\x18\x06 \x01(\x05\"\xf7\x01\n\x12WifiRepeatedAPsGet\x12\x43\n\x0crepeated_aps\x18\x01 \x03(\x0b\x32-.device.api.WifiRepeatedAPsGet.RepeatedAPInfo\x1a\x9b\x01\n\x0eRepeatedAPInfo\x12\x13\n\x0bmac_address\x18\x01 \x01(\t\x12\x0c\n\x04ssid\x18\x02 \x01(\t\x12\"\n\x04\x62\x61nd\x18\x03 \x01(\x0e\x32\x14.device.api.WifiBand\x12\x0f\n\x07\x63hannel\x18\x04 \x01(\r\x12\x0c\n\x04rate\x18\x05 \x01(\r\x12\x0e\n\x06signal\x18\x06 \x01(\x05\x12\x13\n\x0bsignal_bars\x18\x07 \x01(\x05\"\x90\x02\n\x18WifiConnectedStationsGet\x12U\n\x12\x63onnected_stations\x18\x01 \x03(\x0b\x32\x39.device.api.WifiConnectedStationsGet.ConnectedStationInfo\x1a\x9c\x01\n\x14\x43onnectedStationInfo\x12\x13\n\x0bmac_address\x18\x01 \x01(\t\x12)\n\x08vap_type\x18\x02 \x01(\x0e\x32\x17.device.api.WifiVAPType\x12\"\n\x04\x62\x61nd\x18\x03 \x01(\x0e\x32\x14.device.api.WifiBand\x12\x0f\n\x07rx_rate\x18\x04 \x01(\r\x12\x0f\n\x07tx_rate\x18\x05 \x01(\r\"u\n\x19WifiRepeaterParametersSet\x12\x0c\n\x04ssid\x18\x01 \x01(\t\x12\x0b\n\x03key\x18\x02 \x01(\t\x12\x11\n\tcrossband\x18\x03 \x01(\x08\x12*\n\x0cprimary_band\x18\x04 \x01(\x0e\x32\x14.device.api.WifiBand\"K\n!WifiRepeaterParametersSetResponse\x12&\n\x06result\x18\x01 \x01(\x0e\x32\x16.device.api.WifiResult\"9\n\x0fWifiWpsPbcStart\x12&\n\x06result\x18\x01 \x01(\x0e\x32\x16.device.api.WifiResult\"F\n\x1cWifiRepeaterWpsClonePbcStart\x12&\n\x06result\x18\x01 \x01(\x0e\x32\x16.device.api.WifiResult*\xdb\x01\n\nWifiResult\x12\x10\n\x0cWIFI_SUCCESS\x10\x00\x12\x15\n\x11WIFI_INVALID_SSID\x10\x01\x12\x14\n\x10WIFI_INVALID_KEY\x10\x02\x12\x14\n\x10WIFI_IS_DISABLED\x10\x03\x12\x1e\n\x1aWIFI_INVALID_BACKHAUL_SSID\x10\x04\x12\x1d\n\x19WIFI_INVALID_BACKHAUL_KEY\x10\x05\x12 \n\x1cWIFI_UNSUPPORTED_DEVICE_MODE\x10\x06\x12\x17\n\x12WIFI_UNKNOWN_ERROR\x10\xff\x01*E\n\x08WifiBand\x12\x15\n\x11WIFI_BAND_UNKNOWN\x10\x00\x12\x10\n\x0cWIFI_BAND_2G\x10\x01\x12\x10\n\x0cWIFI_BAND_5G\x10\x02*f\n\x0bWifiVAPType\x12\x14\n\x10WIFI_VAP_UNKNOWN\x10\x00\x12\x14\n\x10WIFI_VAP_MAIN_AP\x10\x01\x12\x15\n\x11WIFI_VAP_GUEST_AP\x10\x02\x12\x14\n\x10WIFI_VAP_STATION\x10\x03*^\n\x0bWifiWpaMode\x12\x11\n\rWPA_NO_CHANGE\x10\x00\x12\x0c\n\x08WPA_NONE\x10\x01\x12\x0b\n\x07WPA_1_2\x10\x02\x12\t\n\x05WPA_2\x10\x03\x12\x0b\n\x07WPA_2_3\x10\x04\x12\t\n\x05WPA_3\x10\x05*?\n\x0eWifiDeviceMode\x12\x0f\n\x0bWDM_NO_CONF\x10\x00\x12\n\n\x06WDM_AP\x10\x02\x12\x10\n\x0cWDM_REPEATER\x10\x03\x42\x15\n\x06\x64\x65viceB\x0bwifinetworkb\x06proto3')

_WIFIRESULT = DESCRIPTOR.enum_types_by_name['WifiResult']
WifiResult = enum_type_wrapper.EnumTypeWrapper(_WIFIRESULT)
_WIFIBAND = DESCRIPTOR.enum_types_by_name['WifiBand']
WifiBand = enum_type_wrapper.EnumTypeWrapper(_WIFIBAND)
_WIFIVAPTYPE = DESCRIPTOR.enum_types_by_name['WifiVAPType']
WifiVAPType = enum_type_wrapper.EnumTypeWrapper(_WIFIVAPTYPE)
_WIFIWPAMODE = DESCRIPTOR.enum_types_by_name['WifiWpaMode']
WifiWpaMode = enum_type_wrapper.EnumTypeWrapper(_WIFIWPAMODE)
_WIFIDEVICEMODE = DESCRIPTOR.enum_types_by_name['WifiDeviceMode']
WifiDeviceMode = enum_type_wrapper.EnumTypeWrapper(_WIFIDEVICEMODE)
WIFI_SUCCESS = 0
WIFI_INVALID_SSID = 1
WIFI_INVALID_KEY = 2
WIFI_IS_DISABLED = 3
WIFI_INVALID_BACKHAUL_SSID = 4
WIFI_INVALID_BACKHAUL_KEY = 5
WIFI_UNSUPPORTED_DEVICE_MODE = 6
WIFI_UNKNOWN_ERROR = 255
WIFI_BAND_UNKNOWN = 0
WIFI_BAND_2G = 1
WIFI_BAND_5G = 2
WIFI_VAP_UNKNOWN = 0
WIFI_VAP_MAIN_AP = 1
WIFI_VAP_GUEST_AP = 2
WIFI_VAP_STATION = 3
WPA_NO_CHANGE = 0
WPA_NONE = 1
WPA_1_2 = 2
WPA_2 = 3
WPA_2_3 = 4
WPA_3 = 5
WDM_NO_CONF = 0
WDM_AP = 2
WDM_REPEATER = 3


_WIFIDEVICEMODEGETRESPONSE = DESCRIPTOR.message_types_by_name['WifiDeviceModeGetResponse']
_WIFIDEVICEMODESET = DESCRIPTOR.message_types_by_name['WifiDeviceModeSet']
_WIFIDEVICEMODESETRESPONSE = DESCRIPTOR.message_types_by_name['WifiDeviceModeSetResponse']
_WIFIPARAMETERSSET = DESCRIPTOR.message_types_by_name['WifiParametersSet']
_WIFIPARAMETERSSETRESPONSE = DESCRIPTOR.message_types_by_name['WifiParametersSetResponse']
_WIFIGUESTACCESSSET = DESCRIPTOR.message_types_by_name['WifiGuestAccessSet']
_WIFIGUESTACCESSSETRESPONSE = DESCRIPTOR.message_types_by_name['WifiGuestAccessSetResponse']
_WIFIGUESTACCESSGET = DESCRIPTOR.message_types_by_name['WifiGuestAccessGet']
_WIFINEIGHBORAPSGET = DESCRIPTOR.message_types_by_name['WifiNeighborAPsGet']
_WIFINEIGHBORAPSGET_NEIGHBORAPINFO = _WIFINEIGHBORAPSGET.nested_types_by_name['NeighborAPInfo']
_WIFIREPEATEDAPSGET = DESCRIPTOR.message_types_by_name['WifiRepeatedAPsGet']
_WIFIREPEATEDAPSGET_REPEATEDAPINFO = _WIFIREPEATEDAPSGET.nested_types_by_name['RepeatedAPInfo']
_WIFICONNECTEDSTATIONSGET = DESCRIPTOR.message_types_by_name['WifiConnectedStationsGet']
_WIFICONNECTEDSTATIONSGET_CONNECTEDSTATIONINFO = _WIFICONNECTEDSTATIONSGET.nested_types_by_name['ConnectedStationInfo']
_WIFIREPEATERPARAMETERSSET = DESCRIPTOR.message_types_by_name['WifiRepeaterParametersSet']
_WIFIREPEATERPARAMETERSSETRESPONSE = DESCRIPTOR.message_types_by_name['WifiRepeaterParametersSetResponse']
_WIFIWPSPBCSTART = DESCRIPTOR.message_types_by_name['WifiWpsPbcStart']
_WIFIREPEATERWPSCLONEPBCSTART = DESCRIPTOR.message_types_by_name['WifiRepeaterWpsClonePbcStart']
WifiDeviceModeGetResponse = _reflection.GeneratedProtocolMessageType('WifiDeviceModeGetResponse', (_message.Message,), {
  'DESCRIPTOR' : _WIFIDEVICEMODEGETRESPONSE,
  '__module__' : 'wifinetwork_pb2'
  # @@protoc_insertion_point(class_scope:device.api.WifiDeviceModeGetResponse)
  })
_sym_db.RegisterMessage(WifiDeviceModeGetResponse)

WifiDeviceModeSet = _reflection.GeneratedProtocolMessageType('WifiDeviceModeSet', (_message.Message,), {
  'DESCRIPTOR' : _WIFIDEVICEMODESET,
  '__module__' : 'wifinetwork_pb2'
  # @@protoc_insertion_point(class_scope:device.api.WifiDeviceModeSet)
  })
_sym_db.RegisterMessage(WifiDeviceModeSet)

WifiDeviceModeSetResponse = _reflection.GeneratedProtocolMessageType('WifiDeviceModeSetResponse', (_message.Message,), {
  'DESCRIPTOR' : _WIFIDEVICEMODESETRESPONSE,
  '__module__' : 'wifinetwork_pb2'
  # @@protoc_insertion_point(class_scope:device.api.WifiDeviceModeSetResponse)
  })
_sym_db.RegisterMessage(WifiDeviceModeSetResponse)

WifiParametersSet = _reflection.GeneratedProtocolMessageType('WifiParametersSet', (_message.Message,), {
  'DESCRIPTOR' : _WIFIPARAMETERSSET,
  '__module__' : 'wifinetwork_pb2'
  # @@protoc_insertion_point(class_scope:device.api.WifiParametersSet)
  })
_sym_db.RegisterMessage(WifiParametersSet)

WifiParametersSetResponse = _reflection.GeneratedProtocolMessageType('WifiParametersSetResponse', (_message.Message,), {
  'DESCRIPTOR' : _WIFIPARAMETERSSETRESPONSE,
  '__module__' : 'wifinetwork_pb2'
  # @@protoc_insertion_point(class_scope:device.api.WifiParametersSetResponse)
  })
_sym_db.RegisterMessage(WifiParametersSetResponse)

WifiGuestAccessSet = _reflection.GeneratedProtocolMessageType('WifiGuestAccessSet', (_message.Message,), {
  'DESCRIPTOR' : _WIFIGUESTACCESSSET,
  '__module__' : 'wifinetwork_pb2'
  # @@protoc_insertion_point(class_scope:device.api.WifiGuestAccessSet)
  })
_sym_db.RegisterMessage(WifiGuestAccessSet)

WifiGuestAccessSetResponse = _reflection.GeneratedProtocolMessageType('WifiGuestAccessSetResponse', (_message.Message,), {
  'DESCRIPTOR' : _WIFIGUESTACCESSSETRESPONSE,
  '__module__' : 'wifinetwork_pb2'
  # @@protoc_insertion_point(class_scope:device.api.WifiGuestAccessSetResponse)
  })
_sym_db.RegisterMessage(WifiGuestAccessSetResponse)

WifiGuestAccessGet = _reflection.GeneratedProtocolMessageType('WifiGuestAccessGet', (_message.Message,), {
  'DESCRIPTOR' : _WIFIGUESTACCESSGET,
  '__module__' : 'wifinetwork_pb2'
  # @@protoc_insertion_point(class_scope:device.api.WifiGuestAccessGet)
  })
_sym_db.RegisterMessage(WifiGuestAccessGet)

WifiNeighborAPsGet = _reflection.GeneratedProtocolMessageType('WifiNeighborAPsGet', (_message.Message,), {

  'NeighborAPInfo' : _reflection.GeneratedProtocolMessageType('NeighborAPInfo', (_message.Message,), {
    'DESCRIPTOR' : _WIFINEIGHBORAPSGET_NEIGHBORAPINFO,
    '__module__' : 'wifinetwork_pb2'
    # @@protoc_insertion_point(class_scope:device.api.WifiNeighborAPsGet.NeighborAPInfo)
    })
  ,
  'DESCRIPTOR' : _WIFINEIGHBORAPSGET,
  '__module__' : 'wifinetwork_pb2'
  # @@protoc_insertion_point(class_scope:device.api.WifiNeighborAPsGet)
  })
_sym_db.RegisterMessage(WifiNeighborAPsGet)
_sym_db.RegisterMessage(WifiNeighborAPsGet.NeighborAPInfo)

WifiRepeatedAPsGet = _reflection.GeneratedProtocolMessageType('WifiRepeatedAPsGet', (_message.Message,), {

  'RepeatedAPInfo' : _reflection.GeneratedProtocolMessageType('RepeatedAPInfo', (_message.Message,), {
    'DESCRIPTOR' : _WIFIREPEATEDAPSGET_REPEATEDAPINFO,
    '__module__' : 'wifinetwork_pb2'
    # @@protoc_insertion_point(class_scope:device.api.WifiRepeatedAPsGet.RepeatedAPInfo)
    })
  ,
  'DESCRIPTOR' : _WIFIREPEATEDAPSGET,
  '__module__' : 'wifinetwork_pb2'
  # @@protoc_insertion_point(class_scope:device.api.WifiRepeatedAPsGet)
  })
_sym_db.RegisterMessage(WifiRepeatedAPsGet)
_sym_db.RegisterMessage(WifiRepeatedAPsGet.RepeatedAPInfo)

WifiConnectedStationsGet = _reflection.GeneratedProtocolMessageType('WifiConnectedStationsGet', (_message.Message,), {

  'ConnectedStationInfo' : _reflection.GeneratedProtocolMessageType('ConnectedStationInfo', (_message.Message,), {
    'DESCRIPTOR' : _WIFICONNECTEDSTATIONSGET_CONNECTEDSTATIONINFO,
    '__module__' : 'wifinetwork_pb2'
    # @@protoc_insertion_point(class_scope:device.api.WifiConnectedStationsGet.ConnectedStationInfo)
    })
  ,
  'DESCRIPTOR' : _WIFICONNECTEDSTATIONSGET,
  '__module__' : 'wifinetwork_pb2'
  # @@protoc_insertion_point(class_scope:device.api.WifiConnectedStationsGet)
  })
_sym_db.RegisterMessage(WifiConnectedStationsGet)
_sym_db.RegisterMessage(WifiConnectedStationsGet.ConnectedStationInfo)

WifiRepeaterParametersSet = _reflection.GeneratedProtocolMessageType('WifiRepeaterParametersSet', (_message.Message,), {
  'DESCRIPTOR' : _WIFIREPEATERPARAMETERSSET,
  '__module__' : 'wifinetwork_pb2'
  # @@protoc_insertion_point(class_scope:device.api.WifiRepeaterParametersSet)
  })
_sym_db.RegisterMessage(WifiRepeaterParametersSet)

WifiRepeaterParametersSetResponse = _reflection.GeneratedProtocolMessageType('WifiRepeaterParametersSetResponse', (_message.Message,), {
  'DESCRIPTOR' : _WIFIREPEATERPARAMETERSSETRESPONSE,
  '__module__' : 'wifinetwork_pb2'
  # @@protoc_insertion_point(class_scope:device.api.WifiRepeaterParametersSetResponse)
  })
_sym_db.RegisterMessage(WifiRepeaterParametersSetResponse)

WifiWpsPbcStart = _reflection.GeneratedProtocolMessageType('WifiWpsPbcStart', (_message.Message,), {
  'DESCRIPTOR' : _WIFIWPSPBCSTART,
  '__module__' : 'wifinetwork_pb2'
  # @@protoc_insertion_point(class_scope:device.api.WifiWpsPbcStart)
  })
_sym_db.RegisterMessage(WifiWpsPbcStart)

WifiRepeaterWpsClonePbcStart = _reflection.GeneratedProtocolMessageType('WifiRepeaterWpsClonePbcStart', (_message.Message,), {
  'DESCRIPTOR' : _WIFIREPEATERWPSCLONEPBCSTART,
  '__module__' : 'wifinetwork_pb2'
  # @@protoc_insertion_point(class_scope:device.api.WifiRepeaterWpsClonePbcStart)
  })
_sym_db.RegisterMessage(WifiRepeaterWpsClonePbcStart)

if _descriptor._USE_C_DESCRIPTORS == False:

  DESCRIPTOR._options = None
  DESCRIPTOR._serialized_options = b'\n\006deviceB\013wifinetwork'
  _WIFIRESULT._serialized_start=1875
  _WIFIRESULT._serialized_end=2094
  _WIFIBAND._serialized_start=2096
  _WIFIBAND._serialized_end=2165
  _WIFIVAPTYPE._serialized_start=2167
  _WIFIVAPTYPE._serialized_end=2269
  _WIFIWPAMODE._serialized_start=2271
  _WIFIWPAMODE._serialized_end=2365
  _WIFIDEVICEMODE._serialized_start=2367
  _WIFIDEVICEMODE._serialized_end=2430
  _WIFIDEVICEMODEGETRESPONSE._serialized_start=33
  _WIFIDEVICEMODEGETRESPONSE._serialized_end=102
  _WIFIDEVICEMODESET._serialized_start=105
  _WIFIDEVICEMODESET._serialized_end=274
  _WIFIDEVICEMODESETRESPONSE._serialized_start=276
  _WIFIDEVICEMODESETRESPONSE._serialized_end=343
  _WIFIPARAMETERSSET._serialized_start=345
  _WIFIPARAMETERSSET._serialized_end=391
  _WIFIPARAMETERSSETRESPONSE._serialized_start=393
  _WIFIPARAMETERSSETRESPONSE._serialized_end=460
  _WIFIGUESTACCESSSET._serialized_start=462
  _WIFIGUESTACCESSSET._serialized_end=581
  _WIFIGUESTACCESSSETRESPONSE._serialized_start=583
  _WIFIGUESTACCESSSETRESPONSE._serialized_end=651
  _WIFIGUESTACCESSGET._serialized_start=654
  _WIFIGUESTACCESSGET._serialized_end=784
  _WIFINEIGHBORAPSGET._serialized_start=787
  _WIFINEIGHBORAPSGET._serialized_end=1020
  _WIFINEIGHBORAPSGET_NEIGHBORAPINFO._serialized_start=879
  _WIFINEIGHBORAPSGET_NEIGHBORAPINFO._serialized_end=1020
  _WIFIREPEATEDAPSGET._serialized_start=1023
  _WIFIREPEATEDAPSGET._serialized_end=1270
  _WIFIREPEATEDAPSGET_REPEATEDAPINFO._serialized_start=1115
  _WIFIREPEATEDAPSGET_REPEATEDAPINFO._serialized_end=1270
  _WIFICONNECTEDSTATIONSGET._serialized_start=1273
  _WIFICONNECTEDSTATIONSGET._serialized_end=1545
  _WIFICONNECTEDSTATIONSGET_CONNECTEDSTATIONINFO._serialized_start=1389
  _WIFICONNECTEDSTATIONSGET_CONNECTEDSTATIONINFO._serialized_end=1545
  _WIFIREPEATERPARAMETERSSET._serialized_start=1547
  _WIFIREPEATERPARAMETERSSET._serialized_end=1664
  _WIFIREPEATERPARAMETERSSETRESPONSE._serialized_start=1666
  _WIFIREPEATERPARAMETERSSETRESPONSE._serialized_end=1741
  _WIFIWPSPBCSTART._serialized_start=1743
  _WIFIWPSPBCSTART._serialized_end=1800
  _WIFIREPEATERWPSCLONEPBCSTART._serialized_start=1802
  _WIFIREPEATERWPSCLONEPBCSTART._serialized_end=1872
# @@protoc_insertion_point(module_scope)
