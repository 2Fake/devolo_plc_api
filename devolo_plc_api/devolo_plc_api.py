from aiohttp import ClientSession

from .device_api.device_api import DeviceApi
from .plc_net_api.plc_net_api import PlcNetApi


class DevoloPlcApi():
    def __init__(self, ip: str, session: ClientSession):
        self.device_api = DeviceApi(ip, session)
        self.plc_net_api = PlcNetApi(ip, session)
