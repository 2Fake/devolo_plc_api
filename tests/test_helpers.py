"""Test helper methods."""
from syrupy.assertion import SnapshotAssertion

from devolo_plc_api import wifi_qr_code
from devolo_plc_api.device_api.wifinetwork_pb2 import WPA_2, WifiGuestAccessGet


class TestHelpers:
    """Test devolo_plc_api.helpers."""

    def test_wifi_qr_code(self, snapshot: SnapshotAssertion):
        """Test creating a QR code."""
        wifi_guest_access = WifiGuestAccessGet(enabled=True, ssid='"Test"', key='"Test"', wpa=WPA_2)
        assert wifi_qr_code(wifi_guest_access) == snapshot
