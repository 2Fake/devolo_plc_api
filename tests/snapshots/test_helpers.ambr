# serializer version: 1
# name: TestHelpers.test_wifi_qr_code
  b'<?xml version="1.0" encoding="utf-8"?>\n<svg xmlns="http://www.w3.org/2000/svg" width="37" height="37" class="segno"><path class="qrline" stroke="#000" d="M4 4.5h7m2 0h1m1 0h1m2 0h1m7 0h7m-29 1h1m5 0h1m2 0h1m6 0h2m2 0h1m1 0h1m5 0h1m-29 1h1m1 0h3m1 0h1m2 0h1m1 0h1m2 0h2m1 0h3m2 0h1m1 0h3m1 0h1m-29 1h1m1 0h3m1 0h1m3 0h2m1 0h1m6 0h1m1 0h1m1 0h3m1 0h1m-29 1h1m1 0h3m1 0h1m3 0h1m1 0h1m4 0h1m2 0h1m1 0h1m1 0h3m1 0h1m-29 1h1m5 0h1m1 0h2m5 0h1m1 0h1m1 0h2m1 0h1m5 0h1m-29 1h7m1 0h1m1 0h1m1 0h1m1 0h1m1 0h1m1 0h1m1 0h1m1 0h7m-17 1h2m1 0h3m1 0h1m-20 1h1m2 0h1m1 0h2m1 0h2m4 0h1m1 0h2m3 0h1m1 0h1m-24 1h5m4 0h3m1 0h1m1 0h1m2 0h1m3 0h2m1 0h3m-28 1h1m1 0h1m2 0h2m1 0h1m2 0h4m1 0h1m2 0h2m1 0h2m2 0h2m-28 1h1m1 0h3m3 0h2m1 0h3m5 0h1m1 0h2m2 0h4m-29 1h1m5 0h1m1 0h1m1 0h1m3 0h2m1 0h3m1 0h5m1 0h2m-29 1h5m2 0h1m4 0h3m1 0h2m2 0h1m1 0h1m3 0h3m-29 1h2m4 0h2m1 0h1m2 0h3m3 0h1m2 0h1m4 0h3m-26 1h1m1 0h1m2 0h5m4 0h1m1 0h3m1 0h2m-25 1h2m1 0h1m2 0h3m1 0h1m1 0h2m1 0h2m2 0h1m3 0h1m2 0h1m1 0h1m-28 1h3m1 0h1m1 0h1m1 0h6m3 0h1m2 0h2m3 0h2m-28 1h1m1 0h5m1 0h1m5 0h1m1 0h1m5 0h3m1 0h3m-27 1h1m1 0h1m2 0h3m1 0h2m1 0h1m1 0h2m1 0h1m1 0h1m1 0h1m-24 1h1m1 0h1m3 0h1m1 0h1m1 0h1m1 0h1m3 0h1m3 0h9m-21 1h5m1 0h1m2 0h1m2 0h1m3 0h3m-27 1h7m2 0h2m1 0h1m7 0h1m1 0h1m1 0h2m1 0h1m-28 1h1m5 0h1m1 0h1m5 0h3m1 0h1m1 0h1m3 0h1m1 0h3m-29 1h1m1 0h3m1 0h1m2 0h4m2 0h1m1 0h1m2 0h7m1 0h1m-29 1h1m1 0h3m1 0h1m1 0h1m1 0h1m2 0h1m1 0h1m1 0h3m2 0h2m1 0h1m1 0h1m-28 1h1m1 0h3m1 0h1m2 0h2m3 0h1m1 0h1m1 0h2m1 0h2m1 0h3m1 0h1m-29 1h1m5 0h1m2 0h2m1 0h1m2 0h6m2 0h1m3 0h1m-28 1h7m1 0h1m2 0h1m1 0h1m2 0h2m2 0h1m3 0h1m1 0h2"/></svg>\n'
# ---
